pkg/crc/machine/kubeconfig.go

```diff
@@ -198,13 +198,23 @@ func getTokenForUser(username, password, ip string, ca []byte, clusterConfig *ty
 		},
 	}
 	challengeHandler := challengehandlers.NewBasicChallengeHandler(restConfig.Host, "" /* webconsoleURL */, nil /* in */, nil /* out */, nil /* passwordPrompter */, username, password)
-	token, err := tokenrequest.RequestTokenWithChallengeHandlers(restConfig, challengeHandler)
+	token, err := requestTokenWithChallengeHandlers(restConfig, challengeHandler, ingressHTTPSPort)
 	if err != nil {
 		return "", err
 	}
 	return token, nil
 }
 
+func requestTokenWithChallengeHandlers(clientCfg *restclient.Config, handler *challengehandlers.BasicChallengeHandler, port uint) (string, error) {
+	o, err := tokenrequest.NewRequestTokenOptions(clientCfg, false).WithChallengeHandlers(handler)
+	if err != nil {
+		return "", err
+	}
+
+	o.Issuer = fmt.Sprintf("%s:%d", o.Issuer, port)
```

Comment:
if we are adding the port explicit to issuer then I don't think we really need https://github.com/crc-org/crc/pull/4183/files#diff-a796f5c0a3b2fdd2f85be66107ff80578a3658029bd675c39ac6c015b515ebecR188-R190 part so let's try to remove that and see if that works.